File: preface.py
'''
ID: 22920179
TASK: preface
LANG: PYTHON3
'''
import collections
v = []
v.append(['','I','II','III','IV','V','VI','VII', 'VIII','IX']) # (0)1~9
v.append(['','X','XX','XXX','XL','L','LX','LXX','LXXX','XC']) # (0)10~90
v.append(['','C','CC','CCC','CD','D','DC','DCC','DCCC','CM']) # (0)100~900
v.append(['','M','MM','MMM','','','','','','']) # (0)-1000~3000
str = ''
file1 = open('preface.in','r')
file2 = open('preface.out','w')
n = file1.read()
for k in range(1, int(n)+1):
    i = 0
    while k != 0:
        t = k%10
        str += v[i][t]
        i += 1
        k /= 10
result = collections.Counter(str)
for i in range(0,4):
    for j in range(1, 10):
        if result[v[i][j]] != 0:
            file2.write("%s %d\n" % (v[i][j], result[v[i][j]]))
file1.close()
file2.close()

File: preface.cpp
/*
ID:22920179
PROB:preface
LANG:C++
*/
#include "iostream"
#include "cstdio"
#include "string"
#include "vector"
#include "algorithm"
using namespace std;
int main(){
    freopen("preface.in","r", stdin);
    freopen("preface.out","w", stdout);
    string str[4][10] = {{"","I","II","III","IV","V","VI","VII", "VIII","IX"},
                         {"","X","XX","XXX","XL","L","LX","LXX","LXXX","XC"},
                         {"","C","CC","CCC","CD","D","DC","DCC","DCCC","CM"},
                         {"","M","MM","MMM","","","","","",""}};
    int n; cin >> n;
    string res;
    do{
        int i = 0, t, k=n; 
        while(k != 0){
            t = k%10;
            res += str[i][t];
            i += 1;
            k /= 10;
        }
    }while(0 <-- n);
    char s[] = {'I','V','X','L','C','D','M'};
    for(auto var : s){
        if(count(res.begin(),res.end(), var) != 0){
            cout <<var<<" "<<count(res.begin(),res.end(), var)<<endl;
        }
    }
    return 0;
}